Objective-C/CBLDatabase.mm

```diff
@@ -638,65 +624,29 @@ - (BOOL) createIndexWithConfig: (CBLIndexConfiguration*)config
 }
 
 - (BOOL) createIndex: (NSString*)name withConfig: (id<CBLIndexSpec>)config error: (NSError**)error {
-    CBLAssertNotNil(config);
-    CBLAssertNotNil(name);
-    
-    CBL_LOCK(self) {
-        [self mustBeOpen];
-        
-        CBLStringBytes bName(name);
-        CBLStringBytes c4IndexSpec(config.getIndexSpecs);
-        C4IndexOptions options = config.indexOptions;
-        C4Error c4err;
-        
-        return c4db_createIndex2(_c4db,
-                                 bName,
-                                 c4IndexSpec,
-                                 config.queryLanguage,
-                                 config.indexType,
-                                 &options, &c4err) || convertError(c4err, error);
-    }
+    return [[self mustDefaultCollection: error] createIndexWithName: name
+                                                             config: config
+                                                              error: error];
 }
 
 - (BOOL) deleteIndexForName: (NSString*)name error: (NSError**)outError {
-    CBLAssertNotNil(name);
-    
-    CBL_LOCK(self) {
-        [self mustBeOpen];
-        
-        CBLStringBytes bName(name);
-        C4Error c4err;
-        return c4db_deleteIndex(_c4db, bName, &c4err) || convertError(c4err, outError);
-    }
+    return [[self mustDefaultCollection: outError] deleteIndexWithName: name
+                                                                 error: outError];
 }
 
 #pragma mark - DOCUMENT EXPIRATION
 
 - (BOOL) setDocumentExpirationWithID: (NSString*)documentID
                           expiration: (nullable NSDate*)date
-                               error: (NSError**)error
-{
-    CBLAssertNotNil(documentID);
-    
-    CBL_LOCK(self) {
-        UInt64 timestamp = date ? (UInt64)(date.timeIntervalSince1970*msec) : 0;
-        C4Error err;
-        CBLStringBytes docID(documentID);
-        return c4doc_setExpiration(_c4db, docID, timestamp, &err) || convertError(err, error);
-    }
+                               error: (NSError**)error {
+    return [[self mustDefaultCollection: error] setDocumentExpirationWithID: documentID
+                                                                 expiration: date
+                                                                      error: error];
 }
 
 - (nullable NSDate*) getDocumentExpirationWithID: (NSString*)documentID {
-    CBLAssertNotNil(documentID);
-    
-    CBL_LOCK(self) {
-        CBLStringBytes docID(documentID);
-        UInt64 timestamp = c4doc_getExpiration(_c4db, docID, nullptr);
-        if (timestamp == 0) {
-            return nil;
-        }
-        return [NSDate dateWithTimeIntervalSince1970: (timestamp/msec)];
-    }
+    return [[self mustDefaultCollection: nil] getDocumentExpirationWithID: documentID
```

Comment:
It also needs to check the error if the error is NotOpen or not. If it's NotOpen, an exception needs to be thrown.